lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -2412,7 +2429,7 @@ namespace Js
     {
         if (string->GetLength() > (0 > index || (uint64_t)index + (uint64_t)1 > UINT32_MAX ? UINT32_MAX : (uint32_t)(index + 1)) &&
             NumberUtilities::IsSurrogateLowerPart(string->GetString()[index]) &&
-            NumberUtilities::IsSurrogateUpperPart(string->GetString()[index + 1]))
+            NumberUtilities::IsSurrogateUpperPart(string->GetString()[index + 1]) && isUnicode)
```

Comment:
nit: ideally should do the isUnicode check first - it's much quicker, than the IsSurrogateLowerPart and IsSuurrogateUpperPart checks.